postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresLockDAO.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Conductor Authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.postgres.dao;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.sql.DataSource;
+
+import org.springframework.retry.support.RetryTemplate;
+
+import com.netflix.conductor.core.sync.Lock;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class PostgresLockDAO extends PostgresBaseDAO implements Lock {
+
+    public PostgresLockDAO(
+            RetryTemplate retryTemplate, ObjectMapper objectMapper, DataSource dataSource) {
+        super(retryTemplate, objectMapper, dataSource);
+    }
+
+    @Override
+    public void acquireLock(String lockId) {
+        acquireLock(lockId, Long.MAX_VALUE, Long.MAX_VALUE, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public boolean acquireLock(String lockId, long timeToTry, TimeUnit unit) {
+        return acquireLock(lockId, timeToTry, Long.MAX_VALUE, unit);
+    }
+
+    @Override
+    public boolean acquireLock(String lockId, long timeToTry, long leaseTime, TimeUnit unit) {
+        long endTime = System.currentTimeMillis() + unit.toMillis(timeToTry);
+        while (System.currentTimeMillis() < endTime) {
+            var sql =
+                    "INSERT INTO locks(lock_id, lease_expiration) VALUES (?, timezone('utc', now()) + (?::text || ' milliseconds')::interval) ON CONFLICT (lock_id) DO UPDATE SET lease_expiration = timezone('utc', now()) + (?::text || ' milliseconds')::interval WHERE lock_id = ? AND lease_expiration <= timezone('utc', now())";
```

Comment:
you don't need timezone conversion, now() returns timestamptz

try `select pg_typeof(now()), pg_typeof(timezone('utc', now()));`